Sources/NIOFileSystem/FileSystem.swift

```diff
@@ -1172,6 +1172,27 @@ extension FileSystem {
             )
         }
 
+        #if canImport(Darwin)
+        // COPYFILE_CLONE clones the file if possible and will fallback to doing a copy.
+        // COPYFILE_ALL is shorthand for:
+        //    COPYFILE_STAT | COPYFILE_ACL | COPYFILE_XATTR | COPYFILE_DATA
+        let flags = copyfile_flags_t(COPYFILE_CLONE) | copyfile_flags_t(COPYFILE_ALL)
+        return Libc.copyfile(
+            from: sourcePath,
+            to: destinationPath,
+            state: nil,
+            flags: flags
+        ).mapError { errno in
+            FileSystemError.fcopyfile(
```

Comment:
Can you update this helper? One of the things it'll do is attach a `SyscallError` to the `FileSystemError` which will include the name of the syscall, now it will incorrectly be `fcopyfile`. 

Where it doesn't make sense to duplicate the helper I've tended to just pass in the name of the syscall: `FileSystemError.copyfile("copyfile", ... )`